### PR TITLE
[#4237] test: Add integration tests for tag system

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR proposes to add integration tests for the current tag system, also fixes some corner issues during test.

### Why are the changes needed?

These are necessary e2e tests to make sure tag system is worked.

Fix: #4237 

### Does this PR introduce _any_ user-facing change?

NO.

### How was this patch tested?

ITs to cover.

### CODE REVIEW

**Comment by jerryshao** [comment]
@jerqi @yuqi1129 would you please help to review, thanks.

**Comment by jerryshao** [comment]
@yuqi1129 @jerqi would you please help to review when you have time?

File: api/src/main/java/org/apache/gravitino/tag/SupportsTags.java [hunk 1]
```diff
@@ -58,6 +59,8 @@ public interface SupportsTags {
    * @param tagsToAdd The arrays of tag name to be added to the object.
    * @param tagsToRemove The array of tag name to be removed from the object.
    * @return The array of tag names that are associated with the object.
+   * @throws TagAlreadyAssociatedException If the tag is already associated with the object.
    */
-  String[] associateTags(String[] tagsToAdd, String[] tagsToRemove);
+  String[] associateTags(String[] tagsToAdd, String[] tagsToRemove)
```
Comment (yuqi1129) [inline api/src/main/java/org/apache/gravitino/tag/SupportsTags.java:64]: So, what's the behavior if the tag to be removed does not exist?

Comment (jerryshao) [reply api/src/main/java/org/apache/gravitino/tag/SupportsTags.java:64]: nothing happened.

Comment (jerryshao) [reply api/src/main/java/org/apache/gravitino/tag/SupportsTags.java:68]: Add more comments about this.

**Review by yuqi1129 — COMMENTED** [review]
LGTM

**Review by yuqi1129 — APPROVED** [review]

**Review by mchades — APPROVED** [review]
